Marian tatoeba conversion update

Traubert opened the pull request:
# What does this PR do?

The Helsinki-NLP / Tatoeba NMT models have gone through various
architectural changes, and the old conversion code fails on them. This
commit is something of a rewrite to remedy this, in particular parsing
supplied yaml files rather than README.md files. It needs to be looked
at by someone on the Huggingface side.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?


## Who can review?

@sgugger @sshleifer

Traubert committed:
Update conversion of Tatoeba marian models

The Helsinki-NLP / Tatoeba NMT models have gone through various
architectural changes, and the old conversion code fails on them. This
commit is something of a rewrite to remedy this, in particular parsing
supplied yaml files rather than README.md files. It needs to be looked
at by someone on the Huggingface side.

Traubert committed:
Update conversion of Tatoeba marian models

The Helsinki-NLP / Tatoeba NMT models have gone through various
architectural changes, and the old conversion code fails on them. This
commit is something of a rewrite to remedy this, in particular parsing
supplied yaml files rather than README.md files. It needs to be looked
at by someone on the Huggingface side.

Traubert committed:
Merge branch 'huggingface:master' into marian-tatoeba-conversion-update

Traubert committed:
Use black to reformat code

Traubert committed:
Use black to reformat code

patrickvonplaten committed:
[Flax] Add Beam Search (#12131)

* fix_torch_device_generate_test

* remove @

* push new logit processors

* add processors

* save first working version

* save intermediate

* finish

* make style

* make fix-copies

* finish

* Update tests/test_modeling_flax_bart.py

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Apply suggestions from code review

Co-authored-by: Suraj Patil <surajp815@gmail.com>

Co-authored-by: Patrick von Platen <patrick@huggingface.co>
Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>
Co-authored-by: Suraj Patil <surajp815@gmail.com>

patrickvonplaten committed:
Hubert (#11889)

* fix_torch_device_generate_test

* remove @

* add hubert

* add first test file

* more docs

* fix bugs

* fix bug

* finish

* finish

* finish docstring

* fix

* fix

* finalize

* add to ignored

* finish

* Apply suggestions from code review

* correct naming

* finish

* fix auto config

* finish

* correct convert script

* Apply suggestions from code review

Co-authored-by: Lysandre Debut <lysandre@huggingface.co>
Co-authored-by: Suraj Patil <surajp815@gmail.com>

* apply suggestions lysandre & suraj

Co-authored-by: Lysandre Debut <lysandre@huggingface.co>
Co-authored-by: Suraj Patil <surajp815@gmail.com>

philschmid committed:
updated DLC images and sample notebooks (#12191)

sgugger commented:
cc @patrickvonplaten who did the conversion

sgugger commented on src/transformers/models/marian/convert_marian_tatoeba_to_pytorch.py:
```diff
@@ -46,194 +47,196 @@ class TatoebaConverter:
-        assert opus_readme_path.exists(), f"Readme file {opus_readme_path} not found"
+        model_dir_url = f"{TATOEBA_MODELS_URL}/{model_dict['release']}"
+        long_pair = model_dict["_name"].split("-")
+        assert len(long_pair) == 2
```
No empty assert please, add a small error message.

sgugger commented on src/transformers/models/marian/convert_marian_tatoeba_to_pytorch.py:
```diff
@@ -46,194 +47,196 @@ class TatoebaConverter:
-        # BETTER FRONT MATTER LOGIC
+        langtoken = ""
+        if tgt_multilingual:
+            langtoken = f"* a sentence-initial language token is required in the form of >>id<< (id = valid, usually three-letter target language ID)\n"
```
Nit: can we write this string on several lines to avoid going over the 119 char limit?

sgugger commented on src/transformers/models/marian/convert_marian_tatoeba_to_pytorch.py:
```diff
@@ -46,194 +47,196 @@ class TatoebaConverter:
+
+        # combine with Tatoeba markdown
+        readme_url = f"{TATOEBA_MODELS_URL}/{model_dict['_name']}/README.md"
+        extra_markdown = f"### {model_dict['_name']}\n\n* source name: {self.tag2name[a3_src]} \n* target name: {self.tag2name[a3_tgt]} \n* OPUS readme: [{model_dict['_name']}]({readme_url})\n"
```
Same here.

sgugger reviewed (approved):
Thanks for updating the script! Make sure to run `make style` on your branch to apply the formatting we use for the repo (and remove the red cross in the CI).

Narsil committed:
Enabling AutoTokenizer for HubertConfig. (#12198)

sgugger committed:
Use yaml to create metadata (#12185)

* Use yaml to create metadata

* Fix typo

* Remove pin

bhadreshpsavani committed:
[Docs] fixed broken link (#12205)

* fixed broken link

* Update docs/source/benchmarks.rst

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Update docs/source/benchmarks.rst

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

LysandreJik committed:
Pipeline update & tests (#12207)

patrickvonplaten commented:
Hey @Traubert,

Thanks a lot for adapting the conversion script! Could you maybe post a link to a Marian Tatoeba model so that I can try it out?

NielsRogge committed:
Improve detr (#12147)

* Remove unused variables

* Improve docs

* Fix docs of segmentation masks

Co-authored-by: Lysandre Debut <lysandre@huggingface.co>

sgugger committed:
Add link to the course (#12229)

LysandreJik committed:
Support for torch 1.9.0 (#12224)

* Support for torch 1.9.0

* Torch scatter for 1.9.0

* Github Actions run on 1.9.0

stas00 committed:
fix pt-1.9.0 `add_` deprecation (#12217)

* fix pt-1.9.0 add_ deprecation

* add () for clarity

* Trigger CI

* require_version(torch

LysandreJik committed:
Release: v4.7.0

LysandreJik committed:
Docs for v4.8.0

sgugger committed:
AutoTokenizer: infer the class from the tokenizer config if possible (#12208)

* AutoTokenizer: infer the class from the tokenizer config if possible

* Add tests

* Update src/transformers/models/auto/tokenization_auto.py

Co-authored-by: Patrick von Platen <patrick.v.platen@gmail.com>

Co-authored-by: Patrick von Platen <patrick.v.platen@gmail.com>

bhavitvyamalik committed:
update desc for map in all examples (#12226)

* update desc for map in all examples

* added plm

* suggestions

patil-suraj committed:
[Flax] FlaxAutoModelForSeq2SeqLM (#12228)

* add FlaxAutoModelForSeq2SeqLM

patil-suraj committed:
[FlaxBart] few small fixes (#12247)

* boom boom

* remove flax clip example

* few small fixes

Traubert commented:
I'll fix the issues you mentioned, and I think the output needs to be made a bit neater by omitting some things. Currently a lot of technical details are copied from the model's yaml description. Is there a huggingface guideline for what model cards should look like?

@patrickvonplaten The converter downloads the models, so you should be able to test like:

```python
from convert_marian_tatoeba_to_pytorch import *
conv = TatoebaConverter()
conv.convert_models(('fin-eng',), dry_run = False)
```

This would result in the converter looking in the metadata from the Tatoeba-Challenge repository, which you are supposed to have available locally, and choosing the best model for that pair. It will then download and convert it, I think in that case this file: https://object.pouta.csc.fi/Tatoeba-MT-models/fin-eng/opus+bt-2021-04-30.zip

digantamisra98 committed:
Depreciate pythonic Mish and support PyTorch 1.9 version of Mish (#12240)

* Moved Mish to Torch 1.9 version

* Run black formatting

stas00 committed:
[t5 doc] make the example work out of the box (#12239)

* [run_clm.py] restore caching

* style

* [t5 doc] make the example work out of the box

This PR expands the training example to include the correct model type for the example to work, e.g. with `T5Model` this example will break.

* Update docs/source/model_doc/t5.rst

Co-authored-by: Suraj Patil <surajp815@gmail.com>

* expand the other example

Co-authored-by: Suraj Patil <surajp815@gmail.com>

LysandreJik committed:
Fix the scheduled CI

LysandreJik committed:
Better CI feedback (#12279)

* Better run ID

* Only part of CI

* Revert "Only part of CI"

This reverts commit 29f7f248d21e0f5792e0670ba8705b31ad8967b7.

vishal-burman committed:
Fix for making student ProphetNet for Seq2Seq Distillation (#12130)

* make_student.py: fix to make student ProphetNet

* reformat

patil-suraj committed:
[FlaxClip] fix test from/save pretrained test (#12284)

* boom boom

* remove flax clip example

* fix from_save_pretrained

patil-suraj committed:
[Flax] [WIP] allow loading head model with base model weights (#12255)

* boom boom

* remove flax clip example

* allow loading head model with base model weights

* add test

* fix imports

* disable save, load test for clip

* add test_save_load_to_base

stas00 committed:
[DeepSpeed] don't ignore --adafactor (#12257)

patrickvonplaten committed:
[Flax] Fix flax test save pretrained (#12256)

* fix_torch_device_generate_test

* remove @

* fix flax save pretrained test

Rocketknight1 committed:
Tensorflow QA example (#12252)

* New Tensorflow QA example!

* Style pass

* Updating README.md for the new example

* flake8 fixes

* Update examples/tensorflow/question-answering/README.md

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

patrickvonplaten committed:
[Flax] Add jax flax to env command (#12251)

* fix_torch_device_generate_test

* remove @

* add commands for flax/jax

stas00 committed:
reset report_to to none, avoid deprecation warning (#12293)

stas00 committed:
[trainer + examples] set log level from CLI (#12276)

* set log level from CLI

* add log_level_replica + test + extended docs

* cleanup

* Apply suggestions from code review

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* rename datasets objects to allow datasets module

* improve the doc

* style

* doc improve

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

stas00 committed:
[tests] multiple improvements (#12294)

* [tests] multiple improvements

* cleanup

* style

* todo to investigate

* fix

HamidShojanazeri committed:
Fix for the issue of device-id getting hardcoded for token_type_ids during Tracing [WIP] (#11252)

* registering a buffer for token_type_ids, to pass the error of device-id getting hardcoded when tracing

* sytle format

* adding persistent flag to the resgitered buffers that prevent from adding them to the state_dict and addresses the Backward compatibility issue

* adding the try catch to the fix as persistent flag is only available from PT >1.6

* adding version check

* added the condition to only use the token_type_ids buffer when its autogenerated not passed by user

* adding comments and making the conidtion where token_type_ids are None to use the registered buffer

* taking out position-embeddding from the if block

* adding comments

* handling the case if buffer for position_ids was not registered

* reverted the changes on position_ids, fix the issue with size of token_type_ids buffer, moved the modification for generated token_type_ids to Bertmodel, instead of Embeddings

* reverting the token_type_ids in case of None to the previous version

* reverting changes on position_ids adding back the if block

* changes added by running make fix-copies

* changes added by running make fix-copies and added the import version as it was getting used

* changes added by running make fix-copies

* changes added by running make fix-copies

* fixing the import format

* fixing the import format

* modified to use temp tensor for trimed and expanded token_type_ids buffer

* changes made by fix-copies after temp tensor modifications

* changes made by fix-copies after temp tensor modifications

* changes made by fix-copies after temp tensor modifications

* clean up

* clean up

* clean up

* clean up

* Nit

* Nit

* Nit

* modified according to support device conversion on traced models

* modified according to support device conversion on traced models

* modified according to support device conversion on traced models

* modified according to support device conversion on traced models

* changes based on latest in master

* Adapt templates

* Add version import

Co-authored-by: Ubuntu <ubuntu@ip-172-31-32-81.us-west-2.compute.internal>
Co-authored-by: Lysandre <lysandre.debut@reseau.eseo.fr>

stefan-it committed:
trainer_tf: adjust wandb installation command (#12291)

patil-suraj committed:
add FlaxAutoModelForImageClassification in main init (#12298)

Traubert committed:
Conform to style requirements and clean up some card formatting

Traubert committed:
Conform to style requirements and clean up some card formatting

Traubert committed:
Run isort

Traubert committed:
Run isort

ionicsolutions committed:
Fix and improve documentation for LEDForConditionalGeneration (#12303)

* Replace conditional generation example (fixes #12268)

* Replace model in summarization example with finetuned checkpoint, adapt example text

* Fix typo in new summarization example

* Fix docstring formatting, add missing import statement to example

Traubert commented:
Hm. There's still a failing CI test from isort, but I ran that, committed the change, and on my machine `isort --check-only src/transformers/models/marian/convert_marian_tatoeba_to_pytorch.py` now reports nothing. Any ideas? @sgugger @patrickvonplaten

patrickvonplaten committed:
[Flax] Main doc for event orga (#12305)

* fix_torch_device_generate_test

* remove @

* push

* finish

* some typos

* add more info on communication

* add suggestions

stas00 committed:
[trainer] 2 bug fixes and a rename (#12309)

* bug fixes and a rename

* add extended DDP test

sgugger committed:
FlaxBartPretrainedModel -> FlaxBartPreTrainedModel (#12313)

stas00 committed:
[docs]  performance  (#12258)

* initial performance document

* Apply suggestions from code review

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>
Co-authored-by: Lysandre Debut <lysandre@huggingface.co>

* rewrites based on suggestions

* 8x multiple is for AMP only

* add contribute section

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>
Co-authored-by: Lysandre Debut <lysandre@huggingface.co>

JetRunner committed:
Add CodeCarbon Integration (#12304)

* Add optional dependency

* Add CodeCarbon integration

* Add CodeCarbon integration

* Add CodeCarbon integration

* typo

Narsil committed:
Optimizing away the `fill-mask` pipeline. (#12113)

* Optimizing away the `fill-mask` pipeline.

- Don't send anything to the tokenizer unless needed. Vocab check is
much faster
- Keep BC by sending data to the tokenizer when needed. User handling warning messages will see performance benefits again
- Make `targets` and `top_k` work together better `top_k` cannot be
higher than `len(targets)` but can be smaller still.
- Actually simplify the `target_ids` in case of duplicate (it can happen
because we're parsing raw strings)
- Removed useless code to fail on empty strings. It works only if empty
string is in first position, moved to ignoring them instead.
- Changed the related tests as only the tests would fail correctly
(having incorrect value in first position)

* Make tests compatible for 2 different vocabs... (at the price of a
warning).

Co-authored-by: @EtaoinWu

* ValueError working globally

* Update src/transformers/pipelines/fill_mask.py

Co-authored-by: Lysandre Debut <lysandre@huggingface.co>

* `tokenizer.vocab` -> `tokenizer.get_vocab()` for more compatiblity +
fallback.

Co-authored-by: Lysandre Debut <lysandre@huggingface.co>

stancld committed:
Add output in a dictionary for TF `generate` method (#12139)

* Add output args to greedy search

* Fix critical typo + make style quality

* Handle generate_beam_search

* Add dict_specific tests and fix the placement of encoder outputs

* Add  specific outputs

* Update doc

* Fix typo

* Adjust handling encoder_outputs + Fix generating for T5

* Fix generate for RAG

* Fix handling ouptut_attentions when target_mapping is not None

Take care of situations when target_mapping is provided
as there are 2-tuple of attentions

Change from:
if inputs["output_attentions"]:
    attentions = tuple(tf.transpose(t, perm(2, 3, 0, 1)) for t in attentions)

to:
if inputs["output_attentions"]:
    if inputs["target_mapping"] is not None:
        # when target_mapping is provided, there are 2-tuple of attentions
         attentions = tuple(
             tuple(tf.transpose(attn_stream, perm=(2, 3, 0, 1)) for attn_stream in t) for t in attentions
        )
    else:
        attentions = tuple(tf.transpose(t, perm=(2, 3, 0, 1)) for t in attentions)

* Rename kwargs to model_kwargs

* make style quality

* Move imports in test_modeling_tf_common.py

Move ModelOutput-related imports in test_modeling_tf_common.py
into the `is_tf_available():` statement.

* Rewrite nested if-statements

* Fix added tests

patil-suraj committed:
Flax summarization script  (#12230)

* add summrization script

* fix arguments, preprocessing, metrics

* add generation and metrics

* auto model, prediction loop

* prettify

* label smoothing

* adress Sylvain and Patricks suggestions

* dynamically import shift_tokens_right

* fix shift_tokens_right_fn call

jiafatom committed:
Rewrite ProphetNet to adapt converting ONNX friendly (#11981)

* Rewrite

* [ONNX] rewrite

thevasudevgupta committed:
Flax T5 (#12150)

* copy pytorch-t5

* init

* boom boom

* forward pass same

* make generation work

* add more tests

* make test work

* finish normal tests

* make fix-copies

* finish quality

* correct slow example

* correct slow test

* version table

* upload models

* Update tests/test_modeling_flax_t5.py

* correct incorrectly deleted line

Co-authored-by: Patrick von Platen <patrick.v.platen@gmail.com>
Co-authored-by: Patrick von Platen <patrick@huggingface.co>

LysandreJik committed:
Add mention of the huggingface_hub methods for offline mode (#12320)

patrickvonplaten committed:
[Flax/JAX] Add how to propose projects markdown (#12311)

* fix_torch_device_generate_test

* remove @

* finish

* make style

Commit:
[TFWav2Vec2] Fix docs (#12283)

* fix error

* make style check happy

Co-authored-by: chenhaitao <chenhaitao@qiyi.com>

sgugger committed:
Clean push to hub API (#12187)

* Clean push to hub API

* Create working dir if it does not exist

* Different tweak

* New API + all models + test Flax

* Adds the Trainer clean up

* Update src/transformers/file_utils.py

Co-authored-by: Lysandre Debut <lysandre@huggingface.co>

* Address review comments

* (nit) output types

* No need to set clone_from when folder exists

* Update src/transformers/trainer.py

Co-authored-by: Julien Chaumond <julien@huggingface.co>

* Add generated_from_trainer tag

* Update to new version

* Fixes

Co-authored-by: Lysandre Debut <lysandre@huggingface.co>
Co-authored-by: Julien Chaumond <julien@huggingface.co>
Co-authored-by: Lysandre <lysandre.debut@reseau.eseo.fr>

sgugger committed:
Add all XxxPreTrainedModel to the main init (#12314)

* Add all XxxPreTrainedModel to the main init

* Add to template

* Add to template bis

* Add FlaxT5

LysandreJik committed:
Conda build (#12323)

LysandreJik committed:
Temporarily revert the `fill-mask` improvements.

michaelbenayoun committed:
changed modeling_fx_utils.py to utils/fx.py for clarity (#12326)

Co-authored-by: Michael Benayoun <michael@huggingface.co>

sgugger committed:
Pin good version of huggingface_hub

patrickvonplaten committed:
[Flax T5] Fix weight initialization and fix docs (#12327)

* finish t5 flax fixes

* improve naming

sgugger committed:
Release: v4.8.0

sgugger committed:
v4.9.0.dev0

sam-writer committed:
Update training_args.py (#12328)

mention in `save_strategy` param description that `load_best_model_at_end` can override

stas00 committed:
[Deepspeed] new docs (#12077)

* document sub_group_size

* style

* install + issues reporting

* style

* style

* Update docs/source/main_classes/deepspeed.rst

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* indent 4

* restore

* style

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

sgugger committed:
Fix default to logging_dir lost in merge conflict

richardliaw committed:
try-this (#12338)

Signed-off-by: Richard Liaw <rliaw@berkeley.edu>

patil-suraj committed:
[examples/Flax] move the examples table up (#12341)

LysandreJik committed:
Fix torchscript tests (#12336)

* Fix torchscript tests

* Better test

* Remove bogus print

sgugger committed:
Document patch release v4.8.1

marcvanzee committed:
Add flax/jax quickstart (#12342)

patrickvonplaten committed:
Update README.md

MichalPitr committed:
fixed typo (#12356)

jglaser committed:
Fix exception in prediction loop occurring for certain batch sizes (#12350)

* fix distributed_concat for scalar outputs

* Update README.md

* fixed typo (#12356)

* simplify fix with terser syntax

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* Trigger CI

Co-authored-by: Patrick von Platen <patrick.v.platen@gmail.com>
Co-authored-by: michal pitr <21157924+MichalPitr@users.noreply.github.com>
Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

thevasudevgupta committed:
Add FlaxBigBird QuestionAnswering script (#12233)

* port bigbird script

* adapt script a bit

* change location

* adapt more

* save progress

* init commit

* style

* dataset script tested

* readme add

krfricke committed:
Replace NotebookProgressReporter by ProgressReporter in Ray Tune run (#12357)

* Replace NotebookProgressReporter by ProgressReporter in Ray Tune run

* Move to local import

sgugger committed:
Style

stas00 committed:
remove extra white space from log format (#12360)

cronoik committed:
fixed multiplechoice tokenization (#12362)

* fixed multiplechoice tokenization

The model would have seen two sequences:
1. [CLS]prompt[SEP]prompt[SEP]
2. [CLS]choice0[SEP]choice1[SEP]
that is not correct as we want a contextualized embedding of prompt and choice

* removed outer brackets for proper sequence generation

stas00 committed:
[trainer] add main_process_first context manager (#12351)

* main_process_first context manager

* handle multi-node, add context description

* sync desc

bhadreshpsavani committed:
[Examples] Replicates the new --log_level feature to all trainer-based pytorch (#12359)

* added log_level

* fix comment

* fixed log_level

* Trigger CI

* Unfied logging

* simplified args for log_level

bhadreshpsavani committed:
updated example template (#12365)

bhadreshpsavani committed:
replace print with logger (#12368)

ionicsolutions committed:
[Documentation] Warn that DataCollatorForWholeWordMask is limited to BertTokenizer-like tokenizers (#12371)

* Notify users that DataCollatorForWholeWordMask is limited to BertTokenier-like tokenizers

* Fix code formatting

TahaAslani committed:
Update run_mlm.py (#12344)

Before the code could not be used for validation only because of this line:
extension = data_args.train_file.split(".")[-1]
was assuming that extension must be extracted from the training dataset. This line would run regardless of the training or validation options of the user. This would lead to an error if the user only wants to run an evaluation only and does not want to do train (because the training file does not exist). I modified it to extract extension from the training file if the user wants to do train and extract it from the validation file if the user wants to run eval. This way the code can be used for both training and validation separately.

sgugger committed:
Add possibility to maintain full copies of files (#12312)

stas00 committed:
[CI] add dependency table sync verification (#12364)

* add dependency table sync verification

* improve the message

* improve the message

* revert

* ready to merge

bhadreshpsavani committed:
[Examples] Added context manager to datasets map (#12367)

* added cotext manager to datasets map

* fixed style and spaces

* fixed warning of deprecation

* changed desc

patrickvonplaten committed:
[Flax community event] Add more description to readme (#12398)

* fix_torch_device_generate_test

* remove @

* boom boom

* correct typos

* Apply suggestions from code review

Co-authored-by: Suraj Patil <surajp815@gmail.com>

* Apply suggestions from code review

Co-authored-by: Suzana Ilić <io.suzanai@gmail.com>

* Apply suggestions from code review

Co-authored-by: Suraj Patil <surajp815@gmail.com>
Co-authored-by: Suzana Ilić <io.suzanai@gmail.com>

patrickvonplaten committed:
Update README.md

sgugger committed:
Fix copies

mfuntowicz committed:
Remove the need for `einsum` in Albert's attention computation (#12394)

* debug albert einsum

* Fix matmul computation

* Let's use torch linear layer.

* Style.

patrickvonplaten committed:
[Flax] Adapt flax examples to include `push_to_hub` (#12391)

* fix_torch_device_generate_test

* remove @

* finish

* correct summary writer

* correct push to hub

* fix indent

* finish

* finish

* finish

* finish

* finish

Co-authored-by: Patrick von Platen <patrick@huggingface.co>

Rocketknight1 committed:
Tensorflow LM examples (#12358)

* Tensorflow MLM example

* Add CLM example

* Style fixes, adding missing checkpoint code from the CLM example

* Fix TPU training, avoid massive dataset warnings

* Fix incorrect training length calculation for multi-GPU training

* Fix incorrect training length calculation for multi-GPU training

* Refactors and nitpicks from the review

* Style pass

* Adding README

stas00 committed:
pass the matching trainer log level to deepspeed (#12401)

patrickvonplaten committed:
[Flax] Add T5 pretraining script (#12355)

* fix_torch_device_generate_test

* remove @

* add length computatan

* finish masking

* finish

* upload

* fix some bugs

* finish

* fix dependency table

* correct tensorboard

* Apply suggestions from code review

* correct processing

* slight change init

* correct some more mistakes

* apply suggestions

* improve readme

* fix indent

* Apply suggestions from code review

Co-authored-by: SaulLu <55560583+SaulLu@users.noreply.github.com>

* correct tokenizer

* finish

* finish

* finish

* finish

Co-authored-by: Patrick von Platen <patrick@huggingface.co>
Co-authored-by: SaulLu <55560583+SaulLu@users.noreply.github.com>

stas00 committed:
[models] respect dtype of the model when instantiating it (#12316)

* [models] respect dtype of the model when instantiating it

* cleanup

* cleanup

* rework to handle non-float dtype

* fix

* switch to fp32 tiny model

* improve

* use dtype.is_floating_point

* Apply suggestions from code review

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

* fix the doc

* recode to use explicit torch_dtype_auto_detect, torch_dtype args

* docs and tweaks

* docs and tweaks

* docs and tweaks

* merge 2 args, add docs

* fix

* fix

* better doc

* better doc

Co-authored-by: Sylvain Gugger <35901082+sgugger@users.noreply.github.com>

NielsRogge committed:
Rename detr targets to labels (#12280)

* Rename target to labels in DetrFeatureExtractor

* Update DetrFeatureExtractor tests accordingly

* Improve docs of DetrFeatureExtractor

* Improve docs

* Make style

will-rice committed:
Add out of vocabulary error to ASR models (#12288)

* Add OOV error to ASR models

* Feedback changes

will-rice committed:
Fix TFWav2Vec2 SpecAugment (#12289)

* Fix TFWav2Vec2 SpecAugment

* Invert masks

* Feedback changes

patil-suraj committed:
[example/flax] add summarization readme (#12393)

* add readme

* update readme and add requirements

* Update examples/flax/summarization/README.md

Co-authored-by: Patrick von Platen <patrick.v.platen@gmail.com>

patrickvonplaten committed:
[Flax] Example scripts - correct weight decay  (#12409)

* fix_torch_device_generate_test

* remove @

* finish

* finish

* correct style

hjptriplebee committed:
fix ids_to_tokens naming error in tokenizer of deberta v2 (#12412)

Co-authored-by: Jipeng Huang <jihuan@microsoft.com>

Traubert committed:
Handle "newest" model selection correctly & reformat

Traubert committed:
Handle "newest" model selection correctly & reformat

shamanez committed:
minor fixes in original RAG training (#12395)

sgugger commented:
Make sure you have the good versions installed, they are all pinned so `pip install -e .[quality]` in the repo should do the trick.

Traubert committed:
Merge branch 'marian-tatoeba-conversion-update' of github.com:Traubert/transformers into marian-tatoeba-conversion-update

Traubert committed:
Another attempt at conforming to style

Traubert commented:
> Make sure you have the good versions installed, they are all pinned so `pip install -e .[quality]` in the repo should do the trick.

Thanks - also I didn't realise that `make style` was doing something more than plain `isort`, so now I committed another styled-by-make-style version.

Traubert committed:
flake8 conformance

patrickvonplaten commented:
Sorry for being so unresponsive on this PR @Traubert - do you think it could be possible to open a copy of the PR with a clean git commit history? Think some external git commits got accidentally merged into this PR.
